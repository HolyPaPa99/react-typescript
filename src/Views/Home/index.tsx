import React from 'react';
import { Outlet } from 'react-router';
import { Link } from 'react-router-dom';
import styles from './index.module.scss';

function Home() {
    return (
        <div style={{ display: 'flex' }}>
            <div style={{ marginRight: '20px' }}>
                <h1>Hello, World!</h1>
                <hr />
                <ul className={styles.ul}>
                    <li>
                        <Link to="/">Home</Link>
                    </li>
                    <li>
                        <Link to="page1/1/2">Page1</Link>
                    </li>
                    <li>
                        <Link to="page2">Page2</Link>
                    </li>
                </ul>
            </div>
            <Outlet />
        </div>
    );
}

export default Home;