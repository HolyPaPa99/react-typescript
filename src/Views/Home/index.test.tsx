import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'
import Home from '.'

it('Home', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Home />
        </BrowserRouter>
        ,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
