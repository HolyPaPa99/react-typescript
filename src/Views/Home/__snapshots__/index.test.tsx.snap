// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "marginRight": "20px",
      }
    }
  >
    <h1>
      Hello, World!
    </h1>
    <hr />
    <ul
      className="ul"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/page1/1/2"
          onClick={[Function]}
        >
          Page1
        </a>
      </li>
      <li>
        <a
          href="/page2"
          onClick={[Function]}
        >
          Page2
        </a>
      </li>
    </ul>
  </div>
</div>
`;
